pkg/virt-handler/vm.go

```diff
@@ -1044,12 +1044,17 @@ func (c *VirtualMachineController) updateAccessCredentialConditions(vmi *v1.Virt
 			Message:            message,
 		}
 		vmi.Status.Conditions = append(vmi.Status.Conditions, newCondition)
+
 		if status == k8sv1.ConditionTrue {
 			eventMessage := "Access credentials sync successful."
 			if message != "" {
 				eventMessage = fmt.Sprintf("Access credentials sync successful: %s", message)
 			}
-			c.recorder.Event(vmi, k8sv1.EventTypeNormal, v1.AccessCredentialsSyncSuccess.String(), eventMessage)
+			eventType := k8sv1.EventTypeNormal
+			if domain.Spec.Metadata.KubeVirt.AccessCredential.DeprecatedMethod {
+				eventType = k8sv1.EventTypeWarning
+			}
+			c.recorder.Event(vmi, eventType, v1.AccessCredentialsSyncSuccess.String(), eventMessage)
```

Comment:
There is a `SendK8sEvent` on `Notifier` that could be used. You can pass it through `NewLibvirtDomainManager` down to `AccessCredentialManager`

Reply:
So it would create the event directly from `virt-launcher` ?
I think it would be spammy. In the current code, the event is only shown once, when the condition changes.